parcels/application_kernels/advectiondiffusion.py

```diff
@@ -66,16 +65,18 @@ def AdvectionDiffusionEM(particle, fieldset, time):
     dWx = ParcelsRandom.normalvariate(0, math.sqrt(math.fabs(particle.dt)))
     dWy = ParcelsRandom.normalvariate(0, math.sqrt(math.fabs(particle.dt)))
 
+    u, v = fieldset.UV[time, particle.depth, particle.lat, particle.lon]
+
     Kxp1 = fieldset.Kh_zonal[time, particle.depth, particle.lat, particle.lon + fieldset.dres]
     Kxm1 = fieldset.Kh_zonal[time, particle.depth, particle.lat, particle.lon - fieldset.dres]
     dKdx = (Kxp1 - Kxm1) / (2 * fieldset.dres)
-    ax = fieldset.U[time, particle.depth, particle.lat, particle.lon] + dKdx
+    ax = u + dKdx
```

Comment:
Looks good. Even clearer this way, as it more easily shows what the deterministic term is composed of.